caravel/assets/javascripts/explorev2/reducers/exploreReducer.js

```diff
@@ -2,30 +2,23 @@ import { defaultFormData, defaultOpts } from '../stores/store';
 import * as actions from '../actions/exploreActions';
 import { addToArr, removeFromArr, alterInArr } from '../../../utils/reducerUtils';
 
-const setFormInViz = function (state, action) {
-  const newFormData = Object.assign({}, state);
-  newFormData[action.key] = action.value;
+const setFormInViz = function (oldFd, action) {
+  const newFormData = Object.assign({}, oldFd);
+  // flip the value for checkbox where action.value is undefined
+  newFormData[action.key] = action.value ? action.value : (!oldFd[action.key]);
   return newFormData;
 };
 
-const setVizInState = function (state, action) {
-  switch (action.type) {
-    case actions.SET_FORM_DATA:
-      return Object.assign(
-        {},
-        state,
-        { formData: setFormInViz(state.formData, action) }
-        );
-    default:
-      return state;
-  }
+const setVizInState = function (oldViz, action) {
```

Comment:
i think we should be setting state any time any field is changed, and not just one handler for updating the whole form at once. so we should replace this a function that can set the value for a given field name.